src/Models/BaseElement.php

```diff
@@ -528,6 +528,49 @@ public function getContentForSearchIndex(): string
         return $content;
     }
 
+    /**
+     * Provides content for CMS search
+     */
+    public function getContentForCmsSearch(): string
+    {
+        $contents = [];
+        foreach ($this->config()->get('db') as $fieldName => $fieldType) {
+            // https://docs.silverstripe.org/en/developer_guides/model/data_types_and_casting/
+            // probably won't actually get things like 'CanViewType' in db config
+            if ($fieldName === 'LastEdited'
+                || $fieldName === 'Created'
+                || $fieldName === 'CanViewType'
+                || $fieldName === 'CanEditType'
+                || $fieldName === 'Version'
+                || $fieldName === 'ShowInMenus'
+                || $fieldName === 'ShowInSearch'
+                || $fieldName === 'Sort'
+                || $fieldName === 'HasBrokenFile'
+                || $fieldName === 'HasBrokenLink'
+                || $fieldName === 'ReportClass'
+                || substr($fieldType, -2) === 'ID'
+                || substr($fieldType, -3) === 'Key'
+                || substr($fieldName, -9) === 'ClassName'
+                || substr($fieldType, -4) === 'Hash'
```

Comment:
Unfortunately we cannot use `$searchable_fields` here since the prior "config" was simply whatever was on the rendered template. 

i.e. the generic out-of-the-box `ElementContent` doesn't have `$searchable_fields` https://github.com/silverstripe/silverstripe-elemental/blob/5/src/Models/ElementContent.php

Reply:
> Unfortunately we cannot use $searchable_fields here since the prior "config" was simply whatever was on the rendered template

I don't see how that's a blocker, since this configuration is to explicitly _not_ do that?

> generic out-of-the-box `ElementContent` doesn't have `$searchable_fields`

We can add that, you know. IMO we should just add it to all supported modules which define their own blocks, and from there its up to the developers enabling this opt-in feature to ensure the blocks they're using have the appropriate config.

Reply:
> I don't see how that's a blocker

My point is that fields being rendered in a template currently acts as a quasi config of sorts.  i.e. If you want a field to appear in search, then add it to the template.  No need to add it to proper config.

> We can add that, you know. IMO we should just add it to all supported modules which define their own blocks

Yeah this bits easy.  However we officially support only a few blocks.  

Vast, vast bulk of content blocks out in the wild are defined at the project level, usually as the result of a designer doing UX design for a new website build.   I feel like forcing devs to define to this config which is literally only for CMS page search is adding too much friction and will lead to a worse CMS page search experience since it'll be so easy to forget to do / people won't even know they're supposed to do it - for instance it's pretty normal for junior devs to be tasked with creating new content blocks on a project.

> up to the developers enabling this opt-in feature

It's opt in for CMS 5, ideally I'd like this to be opt-out for CMS 6 because the performance boost is just so high, and CMS page search is probably the single worst performing part of the CMS

Reply:
Have updated to only include Text or Varchar like fields, this greatly limits the amount we need to filter out. We don't  need the InheritedPermissionsExtension or Versioned ones since they only include Enum + Int